src/lib/y2storage/disk_analyzer.rb

```diff
@@ -357,13 +341,28 @@ def all_devices_from_device(device)
       devices.prepend(device)
     end
 
+    # Devices whose name is indicated in the URI of the installation repositories
+    #
+    # @return [Array<BlkDevice>]
+    def repositories_devices
+      return @repositories_devices if @repositories_devices
+
+      devices = repositories_device_names.map { |n| devicegraph.find_by_any_name(n) }.compact
+
+      @repositories_devices = devices
+    end
+
     # Device names indicated in the URI of the installation repositories
     #
     # @see #local_repositories
     #
     # @return [Array<String>]
-    def repositories_devices
-      @repositories_devices ||= local_repositories.map { |r| repository_devices(r) }.flatten
+    def repositories_device_names
+      return @repositories_device_names if @repositories_device_names
+
+      names = local_repositories.map { |r| repository_device_names(r) }.flatten.uniq
```

Comment:
Being nitpicking, maybe flat_map is more efficient (I don't know if it really makes any difference) https://ruby-doc.org/core-3.0.2/Enumerable.html#method-i-flat_map